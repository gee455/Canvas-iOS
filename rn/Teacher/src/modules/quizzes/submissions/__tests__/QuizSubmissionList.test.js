//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

/* eslint-disable flowtype/require-valid-file-annotation */

import React from 'react'
import { QuizSubmissionList, refreshQuizSubmissionData } from '../QuizSubmissionList'
import renderer from 'react-test-renderer'
import cloneDeep from 'lodash/cloneDeep'

const template = {
  ...require('../../../../__templates__/helm'),
  ...require('../../../../__templates__/quiz'),
  ...require('../../../../__templates__/section'),
}

jest.mock('../../../../routing')

const rows = [{
  userID: '1',
  avatarURL: 'http://www.fillmurray.com/200/300',
  name: 'Bill Murray',
  status: 'submitted',
  grade: '8',
  score: 8,
  disclosure: true,
  sectionID: '1',
}]

const props = {
  rows,
  pending: false,
  courseID: '12',
  quizID: template.quiz().id,
  quiz: { data: template.quiz() },
  refreshQuizSubmissions: jest.fn(),
  refreshEnrollments: jest.fn(),
  refreshSections: jest.fn(),
  shouldRefresh: false,
  refreshing: false,
  refresh: jest.fn(),
  sections: [template.section({ id: '1' })],
  getCourseEnabledFeatures: jest.fn(),
}

describe('QuizSubmissionList', () => {
  test('loads correctly when data is supplied', () => {
    const tree = renderer.create(
      <QuizSubmissionList {...props} navigator={template.navigator()} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('loads while pending', () => {
    const tree = renderer.create(
      <QuizSubmissionList {...props} navigator={template.navigator()} pending={true} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('loads with a filter type', () => {
    const tree = renderer.create(
      <QuizSubmissionList {...props} navigator={template.navigator()} filterType='graded' />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('loads with a filter type that does not exist and it should not explode', () => {
    const tree = renderer.create(
      <QuizSubmissionList {...props} navigator={template.navigator()} filterType='whatisthis' />
    ).toJSON()
    expect(tree).toBeDefined()
  })

  test('do not navigate to speedgrader if you do not have an assignment id', () => {
    const navigator = template.navigator()
    const instance = renderer.create(
      <QuizSubmissionList {...props} navigator={navigator} filterType='whatisthis' />
    ).getInstance()
    instance.navigateToSubmission(33)('1')
    expect(navigator.show).not.toHaveBeenCalled()
  })

  test('navigate to speedgrader', () => {
    const localProps = cloneDeep(props)
    localProps.quiz.data.assignment_id = '1'
    const navigator = template.navigator()
    const instance = renderer.create(
      <QuizSubmissionList {...localProps} navigator={navigator} filterType='whatisthis' />
    ).getInstance()
    instance.navigateToSubmission(33)('1')
    expect(navigator.show).toHaveBeenCalledWith(
      '/courses/12/assignments/1/submissions/1',
      { modal: true, modalPresentationStyle: 'fullscreen' },
      { filter: instance.state.filter, studentIndex: 33 }
    )
  })

  test('navigate to submission settings', () => {
    let localProps = cloneDeep(props)
    localProps.quiz.data.assignment_id = '1'
    let navigator = template.navigator()

    let tree = renderer.create(
      <QuizSubmissionList {...localProps} navigator={navigator} filterType='whatisthis' />
    )

    expect(tree.toJSON()).toMatchSnapshot()
    tree.getInstance().openSubmissionSettings()
    expect(navigator.show).toHaveBeenCalledWith(
      `/courses/12/assignments/1/submission_settings`,
      { modal: true },
    )
  })

  test('refresh function', () => {
    refreshQuizSubmissionData(props)
    expect(props.refreshQuizSubmissions).toHaveBeenCalledWith(props.courseID, props.quizID)
    expect(props.refreshEnrollments).toHaveBeenCalledWith(props.courseID)
  })
})
