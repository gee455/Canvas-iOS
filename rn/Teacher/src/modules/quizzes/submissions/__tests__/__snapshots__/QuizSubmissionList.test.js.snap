// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizSubmissionList loads correctly when data is supplied 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "lightgrey",
          "borderBottomWidth": 0.5,
          "flexDirection": "row",
          "height": "auto",
          "justifyContent": "space-between",
          "paddingBottom": 8,
          "paddingHorizontal": 16,
          "paddingTop": 16,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <Text
          accessibilityTraits="header"
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-semibold",
                "fontSize": 20,
              },
              Object {
                "color": "#2d3b44",
                "fontSize": 20,
                "fontWeight": "600",
              },
            ]
          }
        >
          All Submissions
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel="Filter Submissions"
        accessibilityTraits="button"
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={
          Object {
            "bottom": 15,
            "left": 15,
            "right": 15,
            "top": 15,
          }
        }
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginBottom": 1,
            "opacity": 1,
            "width": "auto",
          }
        }
        testID="submission-list.filter"
        tvParallaxProperties={undefined}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "color": "#374A59",
                    "fontSize": 14,
                  },
                  Object {
                    "fontFamily": ".SFUIDisplay-bold",
                  },
                  Object {
                    "color": "#374A59",
                  },
                  Object {},
                ],
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
              ]
            }
          >
            Filter
          </Text>
        </View>
      </View>
    </View>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      ListEmptyComponent={
        <ListEmptyComponent
          title="No results"
        />
      }
      data={
        Array [
          Object {
            "avatarURL": "http://www.fillmurray.com/200/300",
            "disclosure": true,
            "grade": "8",
            "name": "Bill Murray",
            "score": 8,
            "sectionID": "1",
            "status": "submitted",
            "userID": "1",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          progressViewOffset={undefined}
          refreshing={false}
        />
      }
      refreshing={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      testID="quiz-submission-list"
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={
                Array [
                  "button",
                ]
              }
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
              testID="submission-1"
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "flex": 1,
                    "flexDirection": "row",
                    "paddingBottom": 10,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 10,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 40,
                      "marginRight": 8,
                      "width": 40,
                    }
                  }
                >
                  <View
                    accessibilityLabel=""
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Object {
                          "height": 40,
                          "width": 40,
                        },
                        Object {
                          "borderRadius": 20,
                        },
                        Object {
                          "backgroundColor": "#F5F5F5",
                        },
                      ]
                    }
                  >
                    <Image
                      source={
                        Object {
                          "uri": "http://www.fillmurray.com/200/300",
                        }
                      }
                      style={
                        Object {
                          "height": 40,
                          "width": 40,
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "paddingLeft": 8,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    numberOfLines={2}
                    style={
                      Array [
                        Object {
                          "fontFamily": ".SFUIDisplay",
                        },
                        Object {
                          "color": "#2D3B45",
                          "fontSize": 16,
                        },
                        Object {
                          "color": "#2D3B45",
                          "fontSize": 16,
                          "fontWeight": "600",
                        },
                        Object {
                          "fontFamily": ".SFUIDisplay-semibold",
                        },
                      ]
                    }
                  >
                    Bill Murray
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontFamily": ".SFUIDisplay",
                        },
                        Object {
                          "color": "#2D3B45",
                          "fontSize": 16,
                        },
                        Array [
                          Object {
                            "fontSize": 14,
                            "paddingTop": 2,
                          },
                          undefined,
                          Object {
                            "color": "#07AF1F",
                          },
                        ],
                        Object {
                          "fontFamily": ".SFUIDisplay",
                        },
                      ]
                    }
                  >
                    Submitted
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Array [
                        Object {
                          "color": "#374A59",
                          "fontSize": 14,
                          "fontWeight": "600",
                        },
                        Object {
                          "alignSelf": "center",
                        },
                      ],
                      Object {
                        "fontFamily": ".SFUIDisplay-semibold",
                      },
                    ]
                  }
                >
                  8
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`QuizSubmissionList loads while pending 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;

exports[`QuizSubmissionList loads with a filter type 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "lightgrey",
          "borderBottomWidth": 0.5,
          "flexDirection": "row",
          "height": "auto",
          "justifyContent": "space-between",
          "paddingBottom": 8,
          "paddingHorizontal": 16,
          "paddingTop": 16,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <Text
          accessibilityTraits="header"
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-semibold",
                "fontSize": 20,
              },
              Object {
                "color": "#2d3b44",
                "fontSize": 20,
                "fontWeight": "600",
              },
            ]
          }
        >
          Graded
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel="Filter Submissions"
        accessibilityTraits="button"
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={
          Object {
            "bottom": 15,
            "left": 15,
            "right": 15,
            "top": 15,
          }
        }
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginBottom": 1,
            "opacity": 1,
            "width": "auto",
          }
        }
        testID="submission-list.filter"
        tvParallaxProperties={undefined}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "color": "#374A59",
                    "fontSize": 14,
                  },
                  Object {
                    "fontFamily": ".SFUIDisplay-bold",
                  },
                  Object {
                    "color": "#374A59",
                  },
                  Object {},
                ],
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
              ]
            }
          >
            Filter (1)
          </Text>
        </View>
      </View>
    </View>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      ListEmptyComponent={
        <ListEmptyComponent
          title="No results"
        />
      }
      data={
        Array [
          Object {
            "avatarURL": "http://www.fillmurray.com/200/300",
            "disclosure": true,
            "grade": "8",
            "name": "Bill Murray",
            "score": 8,
            "sectionID": "1",
            "status": "submitted",
            "userID": "1",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          progressViewOffset={undefined}
          refreshing={false}
        />
      }
      refreshing={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      testID="quiz-submission-list"
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={
                Array [
                  "button",
                ]
              }
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
              testID="submission-1"
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "flex": 1,
                    "flexDirection": "row",
                    "paddingBottom": 10,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 10,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 40,
                      "marginRight": 8,
                      "width": 40,
                    }
                  }
                >
                  <View
                    accessibilityLabel=""
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Object {
                          "height": 40,
                          "width": 40,
                        },
                        Object {
                          "borderRadius": 20,
                        },
                        Object {
                          "backgroundColor": "#F5F5F5",
                        },
                      ]
                    }
                  >
                    <Image
                      source={
                        Object {
                          "uri": "http://www.fillmurray.com/200/300",
                        }
                      }
                      style={
                        Object {
                          "height": 40,
                          "width": 40,
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "paddingLeft": 8,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    numberOfLines={2}
                    style={
                      Array [
                        Object {
                          "fontFamily": ".SFUIDisplay",
                        },
                        Object {
                          "color": "#2D3B45",
                          "fontSize": 16,
                        },
                        Object {
                          "color": "#2D3B45",
                          "fontSize": 16,
                          "fontWeight": "600",
                        },
                        Object {
                          "fontFamily": ".SFUIDisplay-semibold",
                        },
                      ]
                    }
                  >
                    Bill Murray
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontFamily": ".SFUIDisplay",
                        },
                        Object {
                          "color": "#2D3B45",
                          "fontSize": 16,
                        },
                        Array [
                          Object {
                            "fontSize": 14,
                            "paddingTop": 2,
                          },
                          undefined,
                          Object {
                            "color": "#07AF1F",
                          },
                        ],
                        Object {
                          "fontFamily": ".SFUIDisplay",
                        },
                      ]
                    }
                  >
                    Submitted
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Array [
                        Object {
                          "color": "#374A59",
                          "fontSize": 14,
                          "fontWeight": "600",
                        },
                        Object {
                          "alignSelf": "center",
                        },
                      ],
                      Object {
                        "fontFamily": ".SFUIDisplay-semibold",
                      },
                    ]
                  }
                >
                  8
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`QuizSubmissionList navigate to submission settings 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "lightgrey",
          "borderBottomWidth": 0.5,
          "flexDirection": "row",
          "height": "auto",
          "justifyContent": "space-between",
          "paddingBottom": 8,
          "paddingHorizontal": 16,
          "paddingTop": 16,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <Text
          accessibilityTraits="header"
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-semibold",
                "fontSize": 20,
              },
              Object {
                "color": "#2d3b44",
                "fontSize": 20,
                "fontWeight": "600",
              },
            ]
          }
        >
          All Submissions
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel="Filter Submissions"
        accessibilityTraits="button"
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={
          Object {
            "bottom": 15,
            "left": 15,
            "right": 15,
            "top": 15,
          }
        }
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginBottom": 1,
            "opacity": 1,
            "width": "auto",
          }
        }
        testID="submission-list.filter"
        tvParallaxProperties={undefined}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "color": "#374A59",
                    "fontSize": 14,
                  },
                  Object {
                    "fontFamily": ".SFUIDisplay-bold",
                  },
                  Object {
                    "color": "#374A59",
                  },
                  Object {},
                ],
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
              ]
            }
          >
            Filter
          </Text>
        </View>
      </View>
    </View>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      ListEmptyComponent={
        <ListEmptyComponent
          title="No results"
        />
      }
      data={
        Array [
          Object {
            "avatarURL": "http://www.fillmurray.com/200/300",
            "disclosure": true,
            "grade": "8",
            "name": "Bill Murray",
            "score": 8,
            "sectionID": "1",
            "status": "submitted",
            "userID": "1",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          progressViewOffset={undefined}
          refreshing={false}
        />
      }
      refreshing={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      testID="quiz-submission-list"
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={
                Array [
                  "button",
                ]
              }
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
              testID="submission-1"
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "flex": 1,
                    "flexDirection": "row",
                    "paddingBottom": 10,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 10,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 40,
                      "marginRight": 8,
                      "width": 40,
                    }
                  }
                >
                  <View
                    accessibilityLabel=""
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Object {
                          "height": 40,
                          "width": 40,
                        },
                        Object {
                          "borderRadius": 20,
                        },
                        Object {
                          "backgroundColor": "#F5F5F5",
                        },
                      ]
                    }
                  >
                    <Image
                      source={
                        Object {
                          "uri": "http://www.fillmurray.com/200/300",
                        }
                      }
                      style={
                        Object {
                          "height": 40,
                          "width": 40,
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "paddingLeft": 8,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    numberOfLines={2}
                    style={
                      Array [
                        Object {
                          "fontFamily": ".SFUIDisplay",
                        },
                        Object {
                          "color": "#2D3B45",
                          "fontSize": 16,
                        },
                        Object {
                          "color": "#2D3B45",
                          "fontSize": 16,
                          "fontWeight": "600",
                        },
                        Object {
                          "fontFamily": ".SFUIDisplay-semibold",
                        },
                      ]
                    }
                  >
                    Bill Murray
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontFamily": ".SFUIDisplay",
                        },
                        Object {
                          "color": "#2D3B45",
                          "fontSize": 16,
                        },
                        Array [
                          Object {
                            "fontSize": 14,
                            "paddingTop": 2,
                          },
                          undefined,
                          Object {
                            "color": "#07AF1F",
                          },
                        ],
                        Object {
                          "fontFamily": ".SFUIDisplay",
                        },
                      ]
                    }
                  >
                    Submitted
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Array [
                        Object {
                          "color": "#374A59",
                          "fontSize": 14,
                          "fontWeight": "600",
                        },
                        Object {
                          "alignSelf": "center",
                        },
                      ],
                      Object {
                        "fontFamily": ".SFUIDisplay-semibold",
                      },
                    ]
                  }
                >
                  8
                </Text>
                <View
                  style={
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "transparent",
                      "borderColor": "#c7c7cc",
                      "borderRightWidth": 2,
                      "borderTopWidth": 2,
                      "height": 10,
                      "marginLeft": 7,
                      "transform": Array [
                        Object {
                          "rotate": "45deg",
                        },
                      ],
                      "width": 10,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
